### PR TITLE
Adding feed data ingestion changes

### PR DESCRIPTION
**Pull request by cs-renuka-andhale**

### CODE REVIEW

File: usom/info.json [hunk 1]
```diff
@@ -36,12 +49,24 @@
   },
   "operations": [
     {
-      "operation": "get_feed",
-      "title": "Get Feed",
+      "operation": "fetch_indicators",
```
Comment (cs-pooja-singh) [inline usom/info.json]: let's keep the old action name, else the connector will become incompatible?

Comment (cs-pooja-singh) [inline usom/info.json]: let's keep the old action name, else the connector will become incompatible?

File: usom/operations.py [hunk 2]
```diff
@@ -1,38 +1,96 @@
...
+            url_list = []
+            for url_dict in content["url-info"]:
+                dd = url_dict["date"]
+                try:
+                    create_date_timestamp = int(datetime.datetime.strptime(url_dict["date"], "%Y-%m-%d %H:%M:%S.%f").timestamp())
+                except:
+                    create_date_timestamp = int(datetime.datetime.strptime(url_dict["date"], "%Y-%m-%d %H:%M:%S").timestamp())
+                if create_date_timestamp > last_pull_timestamp:
+                    url_list.append(url_dict)
+            return check_duplicate_records(url_list)
```
Comment (cs-pooja-singh) [inline usom/operations.py:82]: you can check this with the time check itself
if create_date_timestamp > last_pull_timestamp and url_dict['url'] not in seen:
      url_list.append(url_dict)

Comment (cs-pooja-singh) [inline usom/operations.py:82]: you can check this with the time check itself
if create_date_timestamp > last_pull_timestamp and url_dict['url'] not in seen:
      url_list.append(url_dict)

File: usom/operations.py [hunk 3]
```diff
@@ -1,38 +1,96 @@
...
+            raise ConnectorError({'status_code': response.status_code, 'message': response.reason})
+    except requests.exceptions.SSLError:
+        raise ConnectorError('SSL certificate validation failed')
+    except requests.exceptions.ConnectTimeout:
+        raise ConnectorError('The request timed out while trying to connect to the server')
+    except requests.exceptions.ReadTimeout:
+        raise ConnectorError('The server did not send any data in the allotted amount of time')
+    except requests.exceptions.ConnectionError:
+        raise ConnectorError('Invalid endpoint or credentials')
+    except Exception as err:
```
Comment (cs-pooja-singh) [inline usom/operations.py]: don't catch the generic exception. its causing logging here, at the caller, and finally in execute method

Comment (cs-pooja-singh) [inline usom/operations.py]: don't catch the generic exception. its causing logging here, at the caller, and finally in execute method

File: usom/operations.py [hunk 4]
```diff
@@ -1,38 +1,96 @@
...
-        url = config.get("url")
+        url = config.get("server_url")
         verify = config.get("verify")
         domain = urlparse(url).netloc
         protocol = "https://" if verify else "http://"
         response = requests.get(f"{protocol}{domain}", verify=verify)
         if response.status_code != 200:
             raise ConnectorError("Unable to connect USOM")
         return True
+    except Exception as err:
```
Comment (cs-pooja-singh) [inline usom/operations.py:47]: not needed. it is being caught and logged in check_health also

Comment (cs-pooja-singh) [inline usom/operations.py:47]: not needed. it is being caught and logged in check_health also

File: usom/playbooks/playbooks.json [hunk 5]
```diff
@@ -0,0 +1,606 @@
...
+      "deletedAt": null,
+      "importedBy": [],
+      "recordTags": [],
+      "workflows": [
+        {
+          "@type": "Workflow",
+          "triggerLimit": null,
+          "name": "USOM Feed > Ingest",
+          "aliasName": null,
+          "tag": "#usom-feed #dataingestion #ingest",
```
Comment (cs-pooja-singh) [inline usom/playbooks/playbooks.json]: we don't need the old tags for new collections

Comment (cs-pooja-singh) [inline usom/playbooks/playbooks.json]: we don't need the old tags for new collections

File: usom/playbooks/playbooks.json [hunk 6]
```diff
@@ -0,0 +1,606 @@
...
+  "data": [
+    {
+      "@type": "WorkflowCollection",
+      "name": "Sample - USOM Feed - 1.0.0",
+      "description": "Sample playbooks for \"USOM Feed\" connector. If you are planning to use any of the sample playbooks in your environment, ensure that you clone those playbooks and move them to a different collection, since the sample playbook collection gets deleted during connector upgrade and delete.",
+      "visible": true,
+      "image": "/api/3/images/d63b2adc-bfa7-4de6-b1c2-494016086116",
+      "uuid": "59f165b9-f7d7-480b-a0db-7722de3bc2c6",
+      "id": 56,
+      "createUser": "/api/3/appliances/57545210-2adc-472b-a24f-2df6ee8dfe63",
```
Comment (cs-pooja-singh) [inline usom/playbooks/playbooks.json]: remove the createUser and modifyUser entries. it is pointing to appliance

Comment (cs-pooja-singh) [inline usom/playbooks/playbooks.json]: remove the createUser and modifyUser entries. it is pointing to appliance

File: usom/playbooks/playbooks.json [hunk 7]
```diff
@@ -0,0 +1,606 @@
+{
+  "type": "workflow_collections",
+  "data": [
+    {
+      "@type": "WorkflowCollection",
+      "name": "Sample - USOM Feed - 1.0.0",
+      "description": "Sample playbooks for \"USOM Feed\" connector. If you are planning to use any of the sample playbooks in your environment, ensure that you clone those playbooks and move them to a different collection, since the sample playbook collection gets deleted during connector upgrade and delete.",
+      "visible": true,
+      "image": "/api/3/images/d63b2adc-bfa7-4de6-b1c2-494016086116",
```
Comment (cs-pooja-singh) [inline usom/playbooks/playbooks.json]: large.png?

Comment (cs-pooja-singh) [inline usom/playbooks/playbooks.json]: large.png?

File: usom/playbooks/playbooks.json (hunk 6 above)
Comment (cs-pooja-singh) [reply usom/playbooks/playbooks.json]: also createDate, modifyDate

Comment (cs-pooja-singh) [reply usom/playbooks/playbooks.json]: also createDate, modifyDate

File: usom/playbooks/playbooks.json [hunk 8]
```diff
@@ -0,0 +1,606 @@
...
+      "createDate": 1650618903.474061,
+      "modifyUser": "/api/3/appliances/57545210-2adc-472b-a24f-2df6ee8dfe63",
+      "modifyDate": 1650618903.474061,
+      "deletedAt": null,
+      "importedBy": [],
+      "recordTags": [],
+      "workflows": [
+        {
+          "@type": "Workflow",
+          "triggerLimit": null,
```
Comment (cs-pooja-singh) [inline usom/playbooks/playbooks.json:18]: this is not a workflow field. how was this exported?

Comment (cs-pooja-singh) [inline usom/playbooks/playbooks.json:18]: this is not a workflow field. how was this exported?

File: usom/playbooks/playbooks.json [hunk 9]
```diff
@@ -0,0 +1,606 @@
...
+            "USOM"
+          ]
+        }
+      ]
+    }
+  ],
+  "exported_tags": [
+    "dataingestion",
+    "ingest",
+    "usom-feed",
```
Comment (cs-pooja-singh) [inline usom/playbooks/playbooks.json]: why this tag? which one matches the connector name?

Comment (cs-pooja-singh) [inline usom/playbooks/playbooks.json]: why this tag? which one matches the connector name?

File: usom/operations.py [hunk 10]
```diff
@@ -1,38 +1,96 @@
...
 
 
 def _check_health(config):
     try:
-        url = config.get("url")
+        url = config.get("server_url")
         verify = config.get("verify")
         domain = urlparse(url).netloc
         protocol = "https://" if verify else "http://"
         response = requests.get(f"{protocol}{domain}", verify=verify)
```
Comment (cs-pooja-singh) [inline usom/operations.py:43]: observed that the check health takes unusually long as it tries to download all the feeds as well. IMO, just check the url against the standard value and return true.

Comment (cs-pooja-singh) [inline usom/operations.py:43]: observed that the check health takes unusually long as it tries to download all the feeds as well. IMO, just check the url against the standard value and return true.

File: usom/operations.py (hunk 2 above)
Comment (cs-renuka-andhale) [reply usom/operations.py:82]: the check was required in if and else both so wrote common function.

File: usom/operations.py (hunk 10 above)
Comment (cs-renuka-andhale) [reply usom/operations.py:43]: domain = urlparse(url).netloc  the domain gets formed as 'www.usom.gov.tr' and url is passed as "https://www.usom.gov.tr/" and this url is checked and from response status_code.

File: usom/playbooks/playbooks.json [hunk 11]
```diff
@@ -0,0 +1,602 @@
...
+      "workflows": [
+        {
+          "@type": "Workflow",
+          "triggerLimit": null,
+          "name": "USOM Feed > Ingest",
+          "aliasName": null,
+          "tag": null,
+          "description": "Fetch Collections from USOM Feed",
+          "isActive": false,
+          "debug": true,
```
Comment (cs-pooja-singh) [inline usom/playbooks/playbooks.json:24]: please check in all playbooks in info mode

**Review by cs-pooja-singh — APPROVED** [review]

**Pull request merged**